org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/sourcelookup/advanced/JDIHelpers.java

```diff
@@ -75,10 +74,14 @@ public File getClassesLocation(Object element) throws DebugException {
 			try {
 				URL url = new URL(spec);
 				if ("file".equals(url.getProtocol())) { //$NON-NLS-1$
-					return new File(url.toURI()).toPath().normalize().toFile();
+					if (!spec.startsWith("file:/")) { //$NON-NLS-1$
+						// opaque -> URI is not hierarchical https://github.com/eclipse-jdt/eclipse.jdt.debug/issues/330
+						spec = spec.replace("file:", "file:/"); //$NON-NLS-1$ //$NON-NLS-2$
+					}
+					URI uri = URI.create(spec);
```

Comment:
`URL url = new URL(spec);` above is not really used. Should we just replace `if ("file".equals(url.getProtocol()))` with `if(spec.startsWith("file:")`, as we only use `spec` to create URI now?

If yes, we also should first trim the string like `spec = spec.trim();`

Reply:
Or will `url.getProtocol()` also answer "file" for absolute paths like `/hello/my/file` ?

Reply:
that gives java.net.MalformedURLException: no protocol